### DISCUSSION TITLE
How to apply 'template' YAML to new Quarto documents?

### DISCUSSION THREAD

**Discussion by daranzolin** [quarto-dev/quarto-cli#4413] [Q&A]
I have an RStudio project template that creates a Quarto document using `system('quarto create-project {report_dir}')`. I then want to overwrite the YAML header with my preferred defaults. How do I do this? Create a quarto template? Use the ymlthis package?

**Comment by cderv** [comment]
> Create a quarto template?

Using [starter template](https://quarto.org/docs/extensions/starter-templates.html) is probably the official quarto way for such usage. 
Creating [custom project type](https://quarto.org/docs/extensions/project-types.html) is adding another level to embed the custom configuration in a specific extension format. 

Hope this helps

**Reply by daranzolin** [reply]
Thanks @cderv, I think this is a fine solution when working interactively, but I was trying to wrap the `quarto use template` command inside an RStudio project initialization. And when I initialize the project, I get the usual prompt:

```
Quarto templates may execute code when documents are rendered. If you do not 
trust the authors of the template, we recommend that you do not install or 
use the template.
 ? Do you trust the authors of this template (Y/n) »
```

But when I tried to input 'Y', RStudio just spins 'Creating project...' endlessly. I guess I can just manually copy the Quarto template files and accompanying assets from a shared file location, but I was hoping for something more elegant.

**Reply by jimjam-slam** [reply]
`quarto use template` has a `--no-prompt` option that I think is probably designed for interactive use! This might be what you're looking for, @daranzolin 😁

**Reply by daranzolin** [reply]
Thanks @jimjam-slam, I think that might work! Is there an option to specify the directory the files will be downloaded into? When I try to execute the command within my (new) project directory I get:

```
ERROR: Attempting to use a template with '--no-prompt' in a non-empty directory.'
```

**Reply by jimjam-slam** [reply]
Mmmm! When you run the command without `--no-prompt`, it asks you whether you'd like to use a subdirectory _if the current directory is empty_. If you say yes, you're prompted for the subdirectory name (to which it deposits the template), and if you say no, it uses the current directory. If the current directory _isn't_ empty, it doesn't give you the option of using the current directory, skipping straight to the choice of subdirectory name.

My inference from all this is that if you use `--no-prompt`, you're expected to create an empty directory for the command to run in yourself. Maybe that means running something like:

```sh
rm -rf ./templatefolder/* && \
  mkdir -p ./templatefolder && \
  cd templatefolder && \
  quarto use template repo/templatename --no-prompt
```

**Reply by daranzolin** [reply]
I think that's exactly right, in my `initialize_project()` function I ended up doing:

```
dir.create(file.path(path, "Report"))
setwd(file.path(path, "Report"))
system("quarto use template repo/templatename --no-prompt")
```

Thanks!

**Reply by cderv** [reply]
Oh I see. Indeed this is a bit different.  We'll add such function in quarto-r soon. First work on this was : https://github.com/quarto-dev/quarto-r/pull/45

Update to quarto-r should happen soon.
